### PR TITLE
Feat(DatePicker): increase presets prop

### PR DESCRIPTION
**Pull request by CCherry07**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [x] Component style update
- [x] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
增加 presets 属性 预设时间范围快捷选择
> 2. Resolve what problem.
增加 presets 属性 预设时间范围快捷选择
> 3. Related issue link.
#6354
### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
增加usePresets hook 转化 presets 属性值
增加 PresetPanel 组件 展示 presets 参照 antd
> 2. List final API realization and usage sample.
```vue
<template>
    <a-date-picker :presets="presets" />
</template>

<script setup lang="ts">
import { ref } from 'vue'
  const presets =ref([
    { label: 'Yesterday', value: dayjs().add(-1, 'd') },
    { label: 'Last Week', value: dayjs().add(-7, 'd') },
    { label: 'Last Month', value: dayjs().add(-1, 'month') },
  ]);
</script>
```
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
增加新prop
> 2. What will say in changelog?
无
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
The preset ranges for quick selection
> 2. Chinese description (optional)
预设时间范围快捷选择
### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/date-picker/index.en-US.md [hunk 1]
```diff
@@ -95,6 +95,7 @@ The following APIs are shared by DatePicker, RangePicker.
 | placeholder | The placeholder of date input | string \| \[string,string] | - |  |
 | placement | The position where the selection box pops up | `bottomLeft` `bottomRight` `topLeft` `topRight` | bottomLeft | 3.3.0 |
 | popupStyle | To customize the style of the popup calendar | CSSProperties | {} |  |
+| presets | The preset ranges for quick selection | { label: slot, value: [dayjs](https://day.js.org/) }[] | - |  |
```
Comment (Wxh16144) [inline components/date-picker/index.en-US.md:98]: 可以顺手补一个 version 说明从那个版本开始支持的

File: components/date-picker/index.zh-CN.md [hunk 2]
```diff
@@ -97,6 +97,7 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*3OpRQKcygo8AAA
 | placement | 选择框弹出的位置 | `bottomLeft` `bottomRight` `topLeft` `topRight` | bottomLeft | 3.3.0 |
 | popupStyle | 额外的弹出日历样式 | CSSProperties | {} |  |
 | prevIcon | 自定义上一个图标 | slot | - | 3.0 |
+| presets | 预设时间范围快捷选择 | { label: slot, value: [dayjs](https://day.js.org/) }[] | - |  |
```
Comment (Wxh16144) [inline components/date-picker/index.zh-CN.md:100]: 同上

File: components/vc-picker/Picker.tsx [hunk 3]
```diff
@@ -433,7 +439,7 @@ function Picker<DateType>() {
 
       return () => {
         const {
-          prefixCls = 'rc-picker',
+          prefixCls = 'vc-picker',
```
Comment (Wxh16144) [inline components/vc-picker/Picker.tsx:442]: 不做更改

File: components/vc-picker/Picker.tsx [hunk 4]
```diff
@@ -203,6 +208,7 @@ function Picker<DateType>() {
     // ],
     setup(props, { attrs, expose }) {
       const inputRef = ref(null);
+      const presetList = usePresets(props.presets);
```
Comment (tangjinzhou) [inline components/vc-picker/Picker.tsx:211]: 这里丢失响应式了 
usePresets 的参数应该是 ref 或 computedRef

Comment (tangjinzhou) [reply components/vc-picker/Picker.tsx:211]: 同样包括返回的数据也应该是响应式的

File: components/vc-picker/hooks/usePresets.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import warning from 'ant-design-vue/es/vc-util/warning';
+import type { PresetDate } from '../interface';
+
+export default function usePresets<T>(
```
Comment (tangjinzhou) [inline components/vc-picker/hooks/usePresets.ts:4]: 输入、返回 要是响应式的

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/vc-picker/hooks/usePresets.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import type { Ref, ComputedRef } from 'vue';
+import { computed } from 'vue';
+
+import warning from 'ant-design-vue/es/vc-util/warning';
+import type { PresetDate } from '../interface';
+
+export default function usePresets<T>(
```
Comment (zkwolf) [inline components/vc-picker/hooks/usePresets.ts:7]: 用到的地方似乎都只输入了第一个参数？感觉这方法用处不大啊，不如直接删了

**Review by CCherry07 — COMMENTED** [review]
review is ok

File: components/vc-picker/hooks/usePresets.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  presets?: ComputedRef<PresetDate<T>[]>,
+  legacyRanges?: ComputedRef<Record<string, T | (() => T)>>,
+): ComputedRef<PresetDate<T>[]> {
+  if (presets.value) {
```
Comment (tangjinzhou) [inline components/vc-picker/hooks/usePresets.ts:11]: 这样还是有问题的，同样丢失了响应式

Comment (tangjinzhou) [reply components/vc-picker/hooks/usePresets.ts:11]: 我先 merge 修改，你在看下改动

Comment (CCherry07) [reply components/vc-picker/hooks/usePresets.ts:11]: 好的

**Review by tangjinzhou — APPROVED** [review]

File: components/vc-picker/hooks/usePresets.ts (hunk 7 above)
Comment (CCherry07) [reply components/vc-picker/hooks/usePresets.ts:11]:
```ts
      const presets = computed(() => props.presets);
      const ranges = computed(() => props.ranges);
      const presetList = usePresets(presets, ranges);
```
并没有吧，usePresets 返回的是一个计算属性，如果presets.value 存在 就是直接 返回的 presets 计算属性

**Pull request merged**